src/gui/shader_manager.cpp

```diff
@@ -532,6 +532,13 @@ std::string ShaderManager::FindShaderAutoMachine() const
 		return GetCompositeShader();
 	}
 
+	// DOSBOX_RealInit may have not been run yet.
+	// If not, go ahead and set the globals from the config.
+	if (machine == MCH_INVALID) {
```

Comment:
Maybe add a comment that this is a workaround we need to deal with the inconsistent/random/platform & config dependent init order.

Reply:
I believe this problem is going to be consistently reproducible across all platforms.  I'm going to test on Windows here in a little bit to confirm.